docs/developer/architecture/logs.rst

```diff
@@ -0,0 +1,120 @@
+Log Centralisation
+==================
+
+Context
+-------
+
+MetalK8s value is to provide, out of the box, some services in order to ease
+the monitoring and operation of the platform as well as workload running on top
+of it.
+
+MetalK8s provides monitoring service with Prometheus and AlertManager in
+order to expose MetalK8s and workloads metrics and alerts.
+
+The logs generated by the platform and the workloads constitute an essential
+piece of information that can be very helpful when it comes to understand the
+root cause of a failure or a performance degradation. Because of the
+distributed nature of MetalK8s and workloads running of top of it, it is
+important to have some tooling to ease the analysis of near real time and past
+logs from a central endpoint. As such those logs should be stored on the system
+for a configurable period of time. The logs analysis functionalities should be
+accessible through an API and a UI. It is also important to ease correlation
+with other type of indicators such as KPIs and alerts.
+
+The following requirements are not about audit logs but only focus on
+application logs.
+
+Requirements
+------------
+
+- Accessible from a central UI/API
+
+Platform Admin or Storage Admin can visualize logs from all containers from all
+namespaces as well as kubelet, containerd, salt-minion, metalk8s or CLI logs in
+Grafana. One can correlate logs and metrics or alerts in one single Grafana
+Dashboard.
+Browsing logs can be achieved through a documented API.
+
+- Persistence, Retention
+
+Logs should be stored on a persistent storage. Some automatic purging mechanism
+should ensure we are never filling the persistent storage. The Platform
+Administrator can configure the retention period.
+
+- Horizontally scalable (capacity and performance)
+
+The Platform Administrator can add nodes or disk in order to increase storage
+capacity for logs (to keep more history or to store more logs) or to increase
+IOPS. It os also possible to fine tune the deployment configuration in order to
+have more replicas and ideally the Log centralisation service supports
+horizontal POD autoscaling.
+
+- Highly Available
+
+Log collection, storage and browsing services should remain available when
+losing one node or one disk.
+
+- log indexing
+
+For MetalK8s use case we want a lightweight tool to store and expose logs.
+Indexing log on their content provides a powerfull way to analyze logs but it
+will require too much resources both in terms of CPU and Storage. Instead, the
+log centralisation service should offer basic features to group logs.
```

Comment:
Could be awesome if the requirements layout examples of such groupings from the very start.
For example,  log groupings should be based on Container Name or POD Name or Container ID etc

Reply:
done